Improve builder API doc (#13008)
sphinx-doc/sphinx: doc/extdev/builderapi.rst

   .. rubric:: Core Methods

>    These methods are define the core build workflow and cannot be overridden:

Reviewer: bad grammar

```suggestion
   These methods define the core build workflow and should not be overridden:
```
